### PR TITLE
run rustfmt on core/source.rs

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

**Comment by alexcrichton** [comment]
Looks like some of these lines were extended beyond 80 characters, but that's the prevailing style in Cargo. Can you revert back where that happened?

**Comment by sbeckeriv** [comment]
Should the style check file be updated? It says 100 characters. 

https://github.com/rust-lang/cargo/blob/master/tests/check-style.sh

**Comment by alexcrichton** [comment]
Ah perhaps, yeah, although we've tended to have an 80 "soft" limit with a 100 'hard' limit, but I'd be fine making 80 the hard limit.

**Comment by tshepang** [comment]
all but line 414 limited to 80 chars

File: src/cargo/core/source.rs [hunk 1]
```diff
@@ -218,9 +223,8 @@ impl SourceId {
     pub fn with_precise(&self, v: Option<String>) -> SourceId {
         SourceId {
             inner: Arc::new(SourceIdInner {
-                precise: v,
-                .. (*self.inner).clone()
-            }),
+                precise: v, ..(*self.inner).clone()
```
Comment (alexcrichton) [inline src/cargo/core/source.rs]: This should have a newline before the `..`

File: src/cargo/core/source.rs [hunk 2]
```diff
@@ -272,7 +276,7 @@ impl fmt::Display for SourceId {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
         match *self.inner {
             SourceIdInner { kind: Kind::Path, ref url, .. } => {
-                fmt::Display::fmt(url, f)
+                fmt::Display::fmt(url, f),
```
Comment (alexcrichton) [inline src/cargo/core/source.rs]: I don't think this will compile

**Comment by tshepang** [comment]
fixed

**Pull request merged**